cider/common/Arena.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
```

Comment:
直接拷贝过来么？可以thirdparty么？
如果以一定要拷贝过来，这些代码，我理解不应该放在cider下面

Reply:
Several changes have been made as I listed above, and it's not possible to use as thirdparty directly.

Reply:
Should we move these classes one by one with tests ?

Reply:
Yes, it should be, but it will result in a lot of PRs.
Here I migrated them at once and we only need to make some adjustment and add test cases later.

Reply:
if ClickHouse has changed/updated/iterated this codes, and API or performance optimized, how could we insure those changes effect to cider?
maybe we could a create a new repo, only extract ClickHouse‘s basic code we used.

Reply:
Our aim is to make a high performance hashtable as a universal lib, and it's our first step to migrate ck. For now, we have to sync occasionally, and cherry pick commits good for performance. Later if all functions and public APIs are ready, users will use cider instead of any repo else.

Reply:
ok